src/test/scala/strawman/collection/test/TraverseTest.scala

```diff
@@ -1,77 +1,139 @@
+    new BuildFrom[Any, (K, V)] {
+      type To = CC[K, V]
+      def newBuilder(): Builder[(K, V), CC[K, V]] = orderedMapFactory.newBuilder[K, V]
+    }
+
+}
+
+class TraverseTest {
+
+  implicitly[BuildFrom[immutable.List[(Int, String)], (Int, String)]]
+  implicitly[BuildFrom[immutable.TreeMap[Int, String], (Int, String)]]
+
+  def optionSequence[CC[X] <: Iterable[X], A](xs: CC[Option[A]])(implicit bf: BuildFrom[CC[Option[A]], A]): Option[bf.To] =
+    xs.foldLeft[Option[Builder[A, bf.To]]](Some(bf.newBuilder())) {
+      case (Some(builder), Some(a)) => Some(builder += a)
+      case _ => None
+    }.map(_.result)
+
+  def eitherSequence[CC[X] <: Iterable[X], A, B](xs: CC[Either[A, B]])(implicit bf: BuildFrom[CC[Either[A, B]], B]): Either[A, bf.To] =
+    xs.foldLeft[Either[A, Builder[B, bf.To]]](Right(bf.newBuilder())) {
+      case (Right(builder), Right(b)) => Right(builder += b)
+      case (Left(a)       ,        _) => Left(a)
+      case (_             ,  Left(a)) => Left(a)
+    }.right.map(_.result)
+
 //  @Test
-//  def optionSequence1Test: Unit = {
-//    val xs1 = immutable.List(Some(1), None, Some(2))
-//    val o1 = optionSequence1(xs1)
-//    val o1t: Option[immutable.List[Int]] = o1
-//
-//    val xs2 = immutable.TreeSet(Some("foo"), Some("bar"), None)
-//    val o2 = optionSequence1(xs2)
-//    val o2t: Option[immutable.TreeSet[String]] = o2
-//  }
-//
-//  def optionSequenceTest: Unit = {
-//    val xs1 = immutable.List(Some(1), None, Some(2))
-//    val o1 = optionSequence(xs1)
-//    val o1t: Option[immutable.List[Int]] = o1
-//
-//    val xs2 = immutable.TreeSet(Some("foo"), Some("bar"), None)
-//    val o2 = optionSequence(xs2)
-//    val o2t: Option[immutable.TreeSet[String]] = o2
-//
-//    // Breakout-like use case from https://github.com/scala/scala/pull/5233:
-//    val xs4 = immutable.List[Option[(Int, String)]](Some((1 -> "a")), Some((2 -> "b")))
-//    val o4 = optionSequence(xs4)(immutable.TreeMap) // same syntax as in `.to`
-//    val o4t: Option[immutable.TreeMap[Int, String]] = o4
-//  }
-//
-//  @Test
-//  def eitherSequenceTest: Unit = {
-//    val xs3 = mutable.ListBuffer(Right("foo"), Left(0), Right("bar"))
-//    val xs3t: mutable.ListBuffer[Either[Int, String]] = xs3
-//    val e1 = eitherSequence(xs3)
-//    val e1t: Either[Int, mutable.ListBuffer[String]] = e1
-//  }
-//}
+  def optionSequenceTest: Unit = {
+    val xs1 = immutable.List(Some(1), None, Some(2))
+    val o1 = optionSequence(xs1)
+    val o1t: Option[immutable.List[Int]] = o1
+
+    val xs2 = immutable.TreeSet(Some("foo"), Some("bar"), None)
+    val o2 = optionSequence(xs2)
+    val o2t: Option[immutable.TreeSet[String]] = o2
+
+    // Breakout-like use case from https://github.com/scala/scala/pull/5233:
+    val xs4 = immutable.List[Option[(Int, String)]](Some((1 -> "a")), Some((2 -> "b")))
+    val o4 = optionSequence(xs4)(BuildFrom.factory(immutable.TreeMap))
+    val o4t: Option[immutable.TreeMap[Int, String]] = o4
```

Comment:
But it’s possible to explicitly provide a `BuildFrom` instance derived from a collection factory (in such a case, we just ignore the `From` collection, which is a `List[Option[(Int, String)]]` in this example, and use the supplied factory to build the target collection).